src/main/java/com/google/sps/data/Post.java

```diff
@@ -186,34 +252,44 @@ public void entityToPost(Entity entity) {
     timeSort = Integer.parseInt(entity.getProperty("timeSort").toString());
     collegeId = (String) entity.getKind();
     postId = entity.getKey().toString();
+    imageServingUrl = (String) entity.getProperty("imageServingUrl");
+    blobKey = null; // Does not set the blobKey. It can be used to change the blob, so it should not be sent to the client side.
   }
 
-  /* Creates a new entity with the college ID and the Post information. Sets all the properties. */
-  public Entity postToEntity() {
-    Entity newPost = new Entity(collegeId);
+  private BlobKey getBlobKey(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
 
-    newPost.setProperty("organizationName", organizationName);
+    // If the user did not select a file, we cannot generate a blobKey.
+    if(blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
 
-    newPost.setProperty("month", month);
-    newPost.setProperty("day", day);
-    newPost.setProperty("year", year);
+    // The form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
 
-    newPost.setProperty("startHour", startHour);
-    newPost.setProperty("startMinute", startMinute);
-    newPost.setProperty("endHour", endHour);
-    newPost.setProperty("endMinute", endMinute);
+    return blobKey;
+  }
 
-    newPost.setProperty("location", location);
-    newPost.setProperty("lat", lat);
-    newPost.setProperty("lng", lng);
+  private String getServingUrl(BlobKey blobKey) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
 
-    newPost.setProperty("typeOfFood", typeOfFood);
-    newPost.setProperty("numberOfPeopleItFeeds", numberOfPeopleItFeeds);
-    newPost.setProperty("description", description);
+    if (blobKey == null) {
+      return null;
+    }
 
-    newPost.setProperty("timeSort", timeSort);
+    // If the user did not select a file, we cannot generate a URL.
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
```

Comment:
Is there a .isEmpty() function? Not sure. That would be cleaner.

Reply:
There is not! But good idea.